### ISSUE TITLE
Usage with the newly-available warbird.dll

### ISSUE THREAD

**Issue by chocolatkey** [KiFilterFiberContext/warbird-obfuscator#2]
I've been experimenting with using the [newly-available warbird.dll](https://seclists.org/fulldisclosure/2024/Jun/7) to compile a simple C++ application in Visual Studio. I've incorporated the simplest possible warbird config:
```xml
<?xml version="1.0"?>
<Configuration>
    <Debug>
        <CoverInjectedCalls>false</CoverInjectedCalls>
    </Debug>
    <RandomSeed>890372095</RandomSeed>
    <Module>
        <Encryption>
            <Segment SegmentID="101">
                <Include>
                    <Proc>secret</Proc>
                </Include>
            </Segment>
        </Encryption>
    </Module>
</Configuration>
```
and the appropriate flags in the build options, as seen in the attached screenshot:
![warbird](https://github.com/user-attachments/assets/277df7a9-ef50-4d95-ba18-1bfffe168fd2)

However, if I include any modules whatsoever in the warbird config, I get a strange error message, `Warbird failed to transform function ''`
I've simplified my code as much as possible, to just be a main with what is basically an empty body, so I'm not sure what `function ''` is ever referring to.
I'm curious to know if anyone else has had better luck trying this out. If this is not the appropriate place to discuss this, I am also available at <username>@gmail.com or @<username> on Discord

**Comment by AllieTheFox** [comment]
> I've been experimenting with using the [newly-available warbird.dll](https://seclists.org/fulldisclosure/2024/Jun/7) to compile a simple C++ application in Visual Studio. I've incorporated the simplest possible warbird config:
> 
> <?xml version="1.0"?>
> <Configuration>
>     <Debug>
>         <CoverInjectedCalls>false</CoverInjectedCalls>
>     </Debug>
>     <RandomSeed>890372095</RandomSeed>
>     <Module>
>         <Encryption>
>             <Segment SegmentID="101">
>                 <Include>
>                     <Proc>secret</Proc>
>                 </Include>
>             </Segment>
>         </Encryption>
>     </Module>
> </Configuration>
> and the appropriate flags in the build options, as seen in the attached screenshot: ![warbird](https://github.com/user-attachments/assets/277df7a9-ef50-4d95-ba18-1bfffe168fd2)
> 
> However, if I include any modules whatsoever in the warbird config, I get a strange error message, `Warbird failed to transform function ''` I've simplified my code as much as possible, to just be a main with what is basically an empty body, so I'm not sure what `function ''` is ever referring to. I'm curious to know if anyone else has had better luck trying this out. If this is not the appropriate place to discuss this, I am also available at @gmail.com or @ on Discord

apology for very new reply but, any progress or luck since this?

**Comment by chocolatkey** [comment]
@Olive6841 nope, I don’t have the expertise. I still await assistance from someone who knows what the issue is

**Comment by AllieTheFox** [comment]
> [@Olive6841](https://github.com/Olive6841) nope, I don’t have the expertise. I still await assistance from someone who knows what the issue is

I figured out the issue, it required linking the warbird obj files included in the leak however that is unfortunately as far as you can get due to the obj files being arm64 but the warbird.dll being x64 and thus failing to initialize

<img width="1424" height="281" alt="Image" src="https://github.com/user-attachments/assets/b7b0024b-894a-4ea6-a641-d8d3e41cd9b4" />
